### PR TITLE
DTPools: Allow min/max for dtpools range

### PR DESCRIPTION
**Pull request by pavanbalaji**

### CODE REVIEW

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

File: test/mpi/dtpools/src/dtpools.c [hunk 1]
```diff
@@ -116,19 +140,13 @@ int DTP_pool_create_struct(int num_types, MPI_Datatype * basic_types, int *basic
...
     DTPI_OBJ_ALLOC_OR_FAIL(*dtp, sizeof(**dtp));
 
-    /* get number of objects from environment */
-    if ((env_num_objs_str = getenv("DTP_NUM_OBJS"))) {
-        env_num_objs = atoi(env_num_objs_str);
-    }
-
-    num_objs = (env_num_objs < DTPI_OBJ_LAYOUT__STRUCT_NUM && env_num_objs > 0) ?
-        env_num_objs : DTPI_OBJ_LAYOUT__STRUCT_NUM;
+    get_obj_id_range(DTPI_OBJ_LAYOUT__STRUCT_NUM - 1, &min_obj_idx, &max_obj_idx);
```
Comment (gcongiu) [inline test/mpi/dtpools/src/dtpools.c]: There is only one struct layout so `DTPI_OBJ_LAYOUT__STRUCT_NUM` is equal to `1`. Wouldn't this pass a default of `0`?

File: test/mpi/dtpools/src/dtpools.c [hunk 2]
```diff
@@ -46,26 +79,15 @@ int DTP_pool_create(MPI_Datatype basic_type, MPI_Aint basic_type_count, DTP_t *
...
-        num_objs =
-            (env_num_objs < DTPI_OBJ_LAYOUT_SIMPLE__NUM &&
-             env_num_objs > 0) ? env_num_objs : DTPI_OBJ_LAYOUT_SIMPLE__NUM;
-    } else {
-        num_objs =
-            (env_num_objs < DTPI_OBJ_LAYOUT_LARGE__NUM &&
-             env_num_objs > 0) ? env_num_objs : DTPI_OBJ_LAYOUT_LARGE__NUM;
-    }
+    int real_max_idx = basic_type_count == 1 ? DTPI_OBJ_LAYOUT_SIMPLE__NUM - 1 :
+        DTPI_OBJ_LAYOUT_LARGE__NUM - 1;
```
[the remark this answers is not in the document]
Comment (pavanbalaji) [reply test/mpi/dtpools/src/dtpools.c]: It’s the max ID. Do the `-1` should be there.

File: test/mpi/dtpools/src/dtpools.c (hunk 1 above)
Comment (pavanbalaji) [reply test/mpi/dtpools/src/dtpools.c]: Same response as above.

**Comment by pavanbalaji** [comment]
The UCX strict build and the proxy assert are known issues.  I never saw the F90 error before.  Let me investigate a bit.

File: test/mpi/dtpools/src/dtpools.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (gcongiu) [reply test/mpi/dtpools/src/dtpools.c]: Sorry for the confusion I did not notice that `num_objs` accounts for this already

**Comment by pavanbalaji** [comment]
I can't reproduce the F90 error with UCX.  Testing again to see if it's another one of the intermittent errors from UCX.

test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by gcongiu** [comment]
@pavanbalaji UCX seems fine. I will wait for CH3 tests to complete and then merge this if for you is fine.

**Comment by pavanbalaji** [comment]
Tests look good.  This is ready to be signed off and merged.

**Comment by pavanbalaji** [comment]
@gcongiu I made a minor change to the second patch.  Can you take a look at it again?  I've added your signoff to save time, but I can remove it if you have any additional comments.

Retesting with one configuration as a sanity check.

test:jenkins/ch4/ofi

**Comment by pavanbalaji** [comment]
Test looks good.  Just waiting for your approval, Giuseppe.

**Comment by gcongiu** [comment]
@pavanbalaji looks good

**Pull request merged**